index.html

```diff
@@ -604,6 +601,30 @@
   }).then(onSessionStarted);
 });
 
+const matrixUpdateListeners = {};
+function addMatrixUpdateListener(p) {
+  removeMatrixUpdateListener(p);
+
+  const listener = (async function (e) {
+    const matrix = e.data
+    const hash = await this.getHash();
+    channelConnection.setState(this.id, {
+      id: this.id,
+      hash,
+      matrix: matrix.toArray(),
+    });
+  }).bind(p);
```

Comment:
Does this need to be bound if it’s already captured in the closure?

Reply:
No, it doesn't need to be bound. In our other thread it sounded like you wanted "this" to be the package, so I used bind. I can remove it if that's what you're looking for

Reply:
No, I meant that it is probably _already_ bound so we don't need to bind it. This code doesn't look wrong, but this binding here looks like it's not needed.

Reply:
Got it. Since `p` is captured in the closure we can reference `p` in place of `this`. I had that originally and my misunderstanding of your last comment led to this weird/unneeded bind. I'll check in an update soon.